First release review of pipeline (#44)
nf-core/differentialabundance: docs/usage.md

> _Documentation of pipeline parameters is generated automatically from the pipeline schema and can no longer be found in markdown files._

## Introduction

Differential analysis is a common task in a variety of use cases. In essence, all these use cases entail taking an input matrix containing features (e.g. genes) and observations (e.g. samples), and comparing groups of observations in all or a subset of the features. The feature/ observation language here reflects our hope that this workflow will extend in future to encompass a variety of applications where an assumption of gene vs sample may not be a valid one- though that is the application to which the first release will apply.

With the above in mind, running this workflow requires:

- a matrix of quantifications with observations by column and features by row
- a description of the observations such as a sample sheet from RNA-seq analysis
- a description of the features, for our initial RNA-seq application this is simply the GTF file from which gene annotations can be derived
- a specification of how the matrix should be split, and how the resulting groups should be compared

> 

Reviewer: ```suggestion
Currently, this limits the usage of differentialabundance to genomics applications where a GTF annotation file is available. Future developments of the pipeline will encompass changes that enable users to use the pipeline on more generic cases.
```

Author: Thanks! I can't apply the suggestion here, but [adding it to a separate PR against dev](https://github.com/nf-core/differentialabundance/pull/48)

## Observations (samplesheet) input

```bash
--input '[path to samplesheet file]'
```

> This may well be the same sample sheet used to generate the input matrix. For example, in RNA-seq this might be the same sample sheet, perhaps derived from [fetchngs](https://github.com/nf-core/fetchngs), that was input to the [RNA-seq workflow](https://github.com/nf-core/rnaseq). It may be necessary to add columns that describe the groups you want to compare.

Reviewer 2: ```suggestion
The format is very similar to the samplesheet used for [nf-core/RNA-seq](nf-co.re/rnaseq) or as the output of [nf-core/fetchngs](nf-co.re/fetchngs). It may be necessary to add columns that describe the groups you want to compare.
```

Author: This is not what I meant. I meant that this will likely be the same sheet, not just the same format.

TREATED_REP1,AEG588A2_S1_L002_R1_001.fastq.gz,AEG588A2_S1_L002_R2_001.fastq.gz,treated,1
TREATED_REP2,AEG588A2_S1_L003_R1_001.fastq.gz,AEG588A2_S1_L003_R2_001.fastq.gz,treated,2
TREATED_REP3,AEG588A2_S1_L004_R1_001.fastq.gz,AEG588A2_S1_L004_R2_001.fastq.gz,treated,3
```

The file can be tab or comma separated.

## Matrix file

```bash
--matrix '[path to matrix file]'
```

This is a numeric square matrix file, comma or tab-separated, with a column for every observation, and features corresponding to the supplied feature set. The parameters `--observations_id_col` and `--features_id_col` define which of the associated fields should be matched in those inputs.

## Contrasts file

```bash
--contrasts '[path to contrasts file]'
```

The contrasts file references the observations file to define groups of samples to compare. For example, based on the sample sheet above we could define contrasts like:

```console
id,variable,reference,target,blocking
condition_control_treated,condition,control,treated,
condition_control_treated_blockrep,condition,control,treated,replicate
```

The necessary fields in order are:

> - id - an arbitrary identifier, will be used to name contrast-wise output files

Reviewer 2: ```suggestion
- `id` - an arbitrary identifier, will be used to name contrast-wise output files
```
Same for all the other items here

- variable - which column from the observations information will be used to define groups

> - reference - the base/ reference level for the comparison. If features have higer values in this group than target they will generate negative fold changes

Reviewer 2: ```suggestion
- reference - the base/ reference level for the comparison. If features have higher values in this group than target they will generate negative fold changes
```

- target - the target/ non-reference level for the comparison. If features have higher values in this group than the reference they will generate positive fold changes

> - blocking - semicolon-delimited, any additional variables (also observation columns) that should be modelled alongside the contrast variable

Reviewer 2: would be nice to have an example with multiple blocking variables
